### PR TITLE
Add t.me links for User, Chat and Message if available and update User.mention_*

### PR DESCRIPTION
**Pull request by Tr-Jono**
+ `User.link` (t.me link of a user if the user has a username)
+ `Chat.link` (t.me link of a supergroup if it has a username)
+ `Message.link` (t.me link of a message if it is in a supergroup with a username)

Updated `User.mention_markdown` and `User.mention_html` so that the default value of `name` is `User.full_name` instead of `User.name`. If a user has a username, `User.name` returns the user's username prefixed with "@". It makes no sense to markdown a username, since the username already mentions the user.

### CODE REVIEW

File: telegram/chat.py [hunk 1]
```diff
@@ -116,6 +116,15 @@ def __init__(self,
...
         self._id_attrs = (self.id,)
 
+    @property
+    def link(self):
+        """
+        :obj:`str`: Convenience property. If the chat is a supergroup and has a 
+        :attr:`username`, returns a t.me link of the chat.
+        """
+        if self.type == "supergroup" and self.username:
+            return "t.me/{}".format(self.usermame)
```
Comment (bomjacob) [inline telegram/chat.py]: Should this default to something? (if it's not a supergroup)

File: telegram/chat.py [hunk 2]
```diff
@@ -116,6 +116,15 @@ def __init__(self,
         self.bot = bot
         self._id_attrs = (self.id,)
 
+    @property
+    def link(self):
+        """
+        :obj:`str`: Convenience property. If the chat is a supergroup and has a 
+        :attr:`username`, returns a t.me link of the chat.
+        """
+        if self.type == "supergroup" and self.username:
```
Comment (bomjacob) [inline telegram/chat.py]: Please use the Chat.SUPERGROUP attribute here.
(comment applies elsewhere too)

File: telegram/user.py [hunk 3]
```diff
@@ -124,28 +132,24 @@ def de_list(cls, data, bot):
...
-            name (:obj:`str`): If provided, will overwrite the user's name.
-
+            name (:obj:`str`): If provided, will overwrite the user's full name.
         Returns:
             :obj:`str`: The inline mention for the user as markdown.
         """
-        if not name:
-            return util_mention_markdown(self.id, self.name)
-        else:
+        if name:
```
Comment (bomjacob) [inline telegram/user.py]: This change is out of scope... we're more likely to merge simple self contained pull requests...
IMO this is fine this time though :)

Comment (Tr-Jono) [reply telegram/user.py]: Will separate into 2 PRs next time.

Comment (Tr-Jono) [reply telegram/user.py]: I changed the link methods so they return None on default and fixed a mistake I spotted in `Message.link`.

**Comment by Tr-Jono** [comment]
Fixed docstrings. Now working on tests.

**Comment by tsnoam** [comment]
@Tr-Jono 
what is the status of this?

**Comment by Tr-Jono** [comment]
Modified tests for user and chat. Will work on `test_message.py` later.

**Comment by Tr-Jono** [comment]
Tests done!

**Comment by Eldinnie** [comment]
Hi,

As you can see the tests fail. You made some errors in the tests (forgot some comma's some spelling errors.) Please run them localy (run `py.test -v` or `make test` in console) and make sure they pass.

**Comment by Tr-Jono** [comment]
Fixed mistakes, waiting for the checks. I could only work on mobile...

**Comment by Tr-Jono** [comment]
@jeffffc helped fixed the tests & docstrings, and include links for channels. Should be fine now. ☺

**Comment by Eldinnie** [comment]
Hey @Tr-Jono . Did you install the pre-commit hook like explained in our [Contributing guide](https://github.com/python-telegram-bot/python-telegram-bot/blob/master/.github/CONTRIBUTING.rst)

Flake8 found a problem which makes CI fail:

`telegram/user.py:152:1: W293 blank line contains whitespace`

**Comment by jeffffc** [comment]
It was my bad, missed the pre-commit line when I set up a new venv. Will push again a while later, sorry for the silly mistakes we made.

**Comment by Eldinnie** [comment]
No problem :D

**Comment by jeffffc** [comment]
Good to go!

**Comment by Eldinnie** [comment]
i pushed some last fixes and it's good to go with that IMO. The patch fix being 75% is due to missing tests for `mention_html` and `mention_markdown in the first place. Will fix that in another PR.

**Comment by Eldinnie** [comment]
@Tr-Jono one last thing. If you like you can add yourself to the [contributers list](https://github.com/python-telegram-bot/python-telegram-bot/blob/master/AUTHORS.rst).
Just commit or reply here with the entry you want in there and I'll add it when I merge.

**Comment by Eldinnie** [comment]
Thank you for your contribution
